trie/smt.py

```diff
@@ -0,0 +1,177 @@
+from eth_utils import (
+        keccak,
+        to_int,
+    )
+
+from trie.constants import (
+    BLANK_NODE,
+)
+
+from trie.exceptions import (
+        ValidationError,
+    )
+
+from trie.validation import (
+    validate_is_bytes,
+    validate_length,
+)
+
+
+class SparseMerkleTree:
+
+    def __init__(self, keysize=32):
+        # Ensure we can support the given height
+        if 1 > keysize or keysize > 32:
+            raise ValidationError("Keysize must be number of bytes in range [0, 32]")
+
+        self.keysize = keysize  # keysize in bytes
+        self.height = keysize * 8  # height is number of bits in key
```

Comment:
Is `height` a term from the spec.  I think `depth` might be another term we use for this measurement and for me at least it's a more natural name.  I don't feel strongly, but I'd default to pushing for `depth` absent an objective reason to use something else.

Reply:
`depth` :+1: